src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1580,6 +1580,178 @@ LogicalResult ONNXReduceSumV11Op::inferShapes(
+  for (int i = 0; i < spatialRank; ++i) {
+    auto attrSize = ArrayAttrIntVal(kernelShape, i);
+    if (attrSize < 1)
+      return op->emitError("Bad kernel_shape value: must be strictly positive");
+    auto filterSize = filterShape[2 + i];
+    if (filterSize >= 0 && filterSize != attrSize)
+      return op->emitError(
+          "Bad kernel_shape value: does not match filter sizes");
+  }
+  return success();
+}
+
+template <class T>
+static LogicalResult verifyStrides(T *op, int64_t spatialRank) {
+  // 1) Get strides attribute.
+  auto strides = op->strides();
+  if (!strides.hasValue())
+    return success();
+  // 2) Verify that we have the right number.
+  if ((int64_t)ArrayAttrSize(strides) != spatialRank)
+    return op->emitError("strides length incompatible with spatial dimensions");
+  // 3) Verify that they are all positive.
+  for (int i = 0; i < spatialRank; ++i) {
+    auto attrSize = ArrayAttrIntVal(strides, i);
+    if (attrSize < 1)
+      return op->emitError("Bad stride value: must be strictly positive");
+  }
+  return success();
+}
+
+template <class T>
+static LogicalResult verifyDilations(T *op, int64_t spatialRank) {
+  // 1) Get dilation attribute.
+  auto dilations = op->dilations();
+  if (!dilations.hasValue())
+    return success();
+  // 2) verify that we have the right number.
+  if ((int64_t)ArrayAttrSize(dilations) != spatialRank)
+    return op->emitError(
+        "dilations length incompatible with spatial dimensions");
+  // 3) Verify that they are all positive.
+  for (int i = 0; i < spatialRank; ++i) {
+    auto attrSize = ArrayAttrIntVal(dilations, i);
+    if (attrSize < 1)
+      return op->emitError("Bad dilation value: must be strictly positive");
+  }
+  return success();
+}
+
+template <class T>
+static LogicalResult verifyPadding(T *op, int64_t spatialRank) {
+  // Verify auto pad field.
+  auto autoPad = op->auto_pad();
+  if (autoPad == "SAME_UPPER" || autoPad == "SAME_LOWER" ||
+      autoPad == "VALID" || autoPad == "NOTSET") {
+    // Ok, known auto pad value.
+  } else {
+    return op->emitError("Unknown auto pad option");
+  }
+  // Verify pad values, if defined.
+  auto pads = op->pads();
+  if (!pads.hasValue())
+    return success();
+  // Verify that we have the right number of pad values.
+  if ((int32_t)ArrayAttrSize(pads) != 2 * spatialRank)
+    return op->emitError("pads length incompatible with spatial dimensions");
+  // Verify the values of the pads.
+  if (autoPad == "NOTSET") {
+    for (int i = 0; i < 2 * spatialRank; ++i)
+      if (ArrayAttrIntVal(pads, i) < 0)
+        return op->emitError("Bad pad value: must be nonnegative");
+  } else {
+    for (int i = 0; i < 2 * spatialRank; ++i)
+      if (ArrayAttrIntVal(pads, i) != 0)
+        return op->emitError("Bad pad value: nonzero pad value only allowed "
+                             "with NOTSET option");
+  }
+  return success();
+}
+
+static LogicalResult verify(ONNXConvOp op) {
```

Comment:
As I understand this verify will be called when an ONNXConvOp is built. Have you checked that it works if all the inputs of the op are MemRef instead of Tensor?